src/Toolchain.ml

```diff
@@ -274,25 +317,37 @@ let getCommand (t : PackageManager.t) bin args : Path.t * string array =
     | Opam (opam, switch) -> Opam.exec opam ~switch ~args:binArgs
     | Esy (esy, manifest) -> Esy.exec esy ~manifest ~args:binArgs
     | Global -> (Path.ofString bin, Array.of_list args)
+    | Custom template ->
+      let shell = Env.shell () in
+      let command =
+        template
+        |> Js.String.replace "$prog" bin
+        |> Js.String.replace "$args" (String.concat " " args)
+        |> String.trim
+      in
+      let args =
+        match Path.basename (Path.ofString shell) with
+        | "cmd.exe" -> [ "/d"; "/s"; "c"; command ]
+        | _ -> [ "-c"; command ]
+      in
```

Comment:
Instead of launching a process by binary path and argument list, I think we should just use a `ShellExecution` for a custom sandbox.

Reply:
I've been working on something similar to this. But instead of reusing the vscode type, I make a variant type like
```ocaml
type command = 
    | Exec of string
    | Spawn of Path.t * string array
```
Then the custom sandbox can return Exec while the other sandboxes use Spawn. The corresponding `ChildProcess.exec` or `ChildProcess.spawn` function is chosen. I think that should make the command execution simpler.

Reply:
Sounds reasonable to me. I'd call it `Shell` rather than `Exec`.